### PR TITLE
Don't set `tabindex` on panels

### PR DESCRIPTION
**Pull request by koddsson**
From internal issue:

>[...] a tabindex on the full panel is really only appropriate when it entirely consists of text (or the panel is large and the next focusable element is way down in the DOM). But even then, a heading with a tabindex is usually better than a full panel with one.

Ref: https://github.com/github/github/issues/137001

### CODE REVIEW

**Comment by koddsson** [comment]
> In the case of Preview tab, a tabindex is appropriate because it is a panel full of text and we don't control the headings. Hence my suggestion in the thread:
> 
> > With this clarification we might introduce a generic marker for tabpanels that has a primary focusable area to avoid tabindex being set to the panel.

I saw that and I read the whole thread as it was up to implementer to make sure the panel had a focusable area. So in the preview tab should just have a `tabindex="0"` in the markup. 

Could you elborate on what a "general marker" is in this context?

File: index.js [hunk 1]
```diff
@@ -59,11 +59,9 @@ function selectTab(tabContainer: TabContainerElement, index: number) {
 
   for (const tab of tabs) {
     tab.setAttribute('aria-selected', 'false')
-    tab.setAttribute('tabindex', '-1')
```
Comment (muan) [inline index.js]: This looks like an accidental change?

Comment (koddsson) [reply index.js]: Oh yeah 😄

File: index.js [hunk 2]
```diff
@@ -63,7 +63,9 @@ function selectTab(tabContainer: TabContainerElement, index: number) {
   }
   for (const panel of panels) {
     panel.hidden = true
-    panel.setAttribute('tabindex', '0')
+    if (!panel.hasAttribute('data-tab-container-no-tabstop')) {
```
Comment (muan) [inline index.js]:
```suggestion
    if (!panel.hasAttribute('tabindex') && !panel.hasAttribute('data-tab-container-no-tabstop')) {
```

**Pull request merged**